pkg/predicate/predicate_test.go

```diff
@@ -308,4 +308,134 @@ var _ = Describe("Predicate", func() {
 		})
 
 	})
+
+	Describe("When checking a GenerationChangedPredicate", func() {
+		instance := predicate.GenerationChangedPredicate{}
+		Context("Where the old object doesn't have a Generation or metadata", func() {
+			It("should return false", func() {
+				new := &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:       "baz",
+						Namespace:  "biz",
+						Generation: 1,
+					}}
+
+				failEvnt := event.UpdateEvent{
+					MetaNew:   new.GetObjectMeta(),
+					ObjectNew: new,
+				}
+				Expect(instance.Create(event.CreateEvent{})).Should(BeTrue())
```

Comment:
`Expect` notation should be used with `To` instead of `Should` https://onsi.github.io/gomega/#getting-gomega to be idiomatic. I think our older code has this problem, but we can fix this for newer tests.

`Expect(instance.Create(event.CreateEvent{})).To(BeTrue())`

This comment applies to all the `Expect` notations in the tests below. [avoiding typing ditto :)]